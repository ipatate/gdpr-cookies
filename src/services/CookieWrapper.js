// @flow
import typeof Cookie from 'js-cookie';
import JSCookie from 'js-cookie';

class CookieWrapper {
  cookie: Cookie;

  constructor() {
    this.cookie = JSCookie;
  }
  /**
   * @description get all cookies
   * @return {object}
   */
  getAll(): Object {
    return this.cookie.get();
  }

  /**
   * @description get cookie by name
   * @param {string} name
   * @return {string | void}
   */
  get(name: string): string | void {
    return this.cookie.get(name);
  }

  /**
   * @description set cookie by name and value
   * @param {string} name
   * @param {string} value
   * @return {void}
   */
  set(name: string, value: string, expires: ?number): void {
    let options = {};
    if (expires) options = {expires};
    return this.cookie.set(name, value, options);
  }

  /**
   * @description remove cookie by name
   * @param {string} name
   * @return {void}
   */
  remove(name: string): void {
    return this.cookie.remove(name);
  }
}

export default CookieWrapper;
