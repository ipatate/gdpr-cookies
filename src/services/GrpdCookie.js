// @flow
import CookieWrapper from './CookieWrapper';

export default class GrpdCookie {
  name: string;
  cookie: CookieWrapper;

  /**
   * @description constructor !
   * @param {object} options
   * @return {void}
   */
  constructor(name: string = 'grpd_cookie'): void {
    this.name = name;
    this.cookie = new CookieWrapper();
  }

  /**
   * @description get cookie with grpd settings
   * @return {string|void}
   */
  getCookie(): Object | void {
    const stringifyValue = this.cookie.get(this.name);
    return stringifyValue !== undefined
      ? JSON.parse(stringifyValue)
      : stringifyValue;
  }

  /**
   * @description update grpd setting in cookie
   * @param {string} value to insert in cookie
   * @return {string}
   */
  updateCookie(value: Object): ?string {
    const stringifyValue = JSON.stringify(value);
    return this.cookie.set(this.name, stringifyValue);
  }

  /**
   * @description remove cookie exclude name in cookie array and name of cookie rgpd
   * @param {array} keepCookie
   * @return {void}
   */
  removeAll(keepCookie: Array<string> = []): void {
    const cookies = this.cookie.getAll();
    for (const cookie in cookies) {
      if (keepCookie.indexOf(cookie) === -1 && cookie !== this.name) {
        this.cookie.remove(cookie);
      }
    }
  }
}
