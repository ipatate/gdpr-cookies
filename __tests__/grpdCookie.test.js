import CookieWrapper from '../src/services/CookieWrapper';
import GrpdCookie from '../src/services/GrpdCookie';

test('get exist value cookie', () => {
  const grpdCookie = new GrpdCookie();
  const cookieWrapper = new CookieWrapper();
  cookieWrapper.set(grpdCookie.name, {foo: 'bar'});
  expect(grpdCookie.getCookie()).toMatchSnapshot();
});

test('update value cookie', () => {
  const grpdCookie = new GrpdCookie();
  const cookieWrapper = new CookieWrapper();
  grpdCookie.updateCookie({lol: true, plop: false});
  expect(cookieWrapper.get(grpdCookie.name)).toMatchSnapshot();
});

test('get not exist value', () => {
  const grpdCookie = new GrpdCookie();
  const cookieWrapper = new CookieWrapper();

  const optionName = grpdCookie.name;
  cookieWrapper.remove(optionName);
  expect(grpdCookie.getCookie()).toBeUndefined();
});
