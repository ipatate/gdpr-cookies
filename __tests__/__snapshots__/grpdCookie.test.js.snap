// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get exist value cookie 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`update value cookie 1`] = `"{\\"lol\\":true,\\"plop\\":false}"`;
